### PR TITLE
Replace Bytes with a List of UInt8

### PR DESCRIPTION
**Pull request by OceanOak**
Changelog:

```
Area of change
- Replace the `Bytes` type with a `List<UInt8>`
```

### CODE REVIEW

File: backend/src/BuiltinExecution/Libs/Base64.fs [hunk 1]
```diff
@@ -81,22 +87,54 @@ let fns : List<BuiltInFn> =
...
           System.Convert.ToBase64String(bytes).Replace('+', '-').Replace('/', '_')
           |> DString
           |> Ply
         | _ -> incorrectArgs ())
       sqlSpec = NotYetImplemented
       previewable = Pure
+      deprecated = NotDeprecated }
+
+
+    { name = fn "hexEncode" 0
```
Comment (OceanOak) [inline backend/src/BuiltinExecution/Libs/Base64.fs:110]: Not sure if this is the right place for this function. It was originally in 'Bytes.fs' and is used in both crypto and big tests

File: backend/src/LibExecution/Dval.fs [hunk 2]
```diff
@@ -104,3 +104,18 @@ let result
   match dv with
   | Ok dv -> resultOk okType errorType dv
   | Error dv -> resultError okType errorType dv
+
+
+let byteArrayToDvalList (bytes : byte[]) : Dval =
```
Comment (OceanOak) [inline backend/src/LibExecution/Dval.fs:109]: Not sure if this is the right place for these two helper functions

File: backend/tests/Tests/DvalRepr.Tests.fs [hunk 3]
```diff
@@ -99,28 +99,29 @@ let testToDeveloperRepr =
...
+//         Expect.equal actual expected "bad fsharp impl"
+//       }
 
-    testList
-      "hashv2"
-      [ t (NEList.singleton (DBytes [||])) "X1YnxJLFsVg"
-        t (NEList.singleton (DBytes [| 128uy |])) "Hj0nqyrvXis" ]
+//     testList
+//       "hashv2"
+//       [ t (NEList.singleton (DList(VT.uint8, []))) "X1YnxJLFsVg" //DEux3mJnJPs
```
Comment (OceanOak) [inline backend/tests/Tests/DvalRepr.Tests.fs:121]: Is it expected to get different hash values here?

Comment (pbiggar) [reply backend/tests/Tests/DvalRepr.Tests.fs:121]: yeah, I'd expect that.

File: backend/src/LibExecution/Dval.fs (hunk 2 above)
Comment (pbiggar) [reply backend/src/LibExecution/Dval.fs:109]: This is where I would put them too

File: backend/src/BuiltinExecution/Libs/Base64.fs (hunk 1 above)
Comment (pbiggar) [reply backend/src/BuiltinExecution/Libs/Base64.fs:109]: I'd maybe put these in bytes? It feels like this could go into a package though?

Comment (OceanOak) [reply backend/src/BuiltinExecution/Libs/Base64.fs]: I moved them to Bytes for now, I'll see if I can move it to a package

File: backend/src/BuiltinExecution/Libs/String.fs [hunk 4]
```diff
@@ -395,14 +394,14 @@ let fns : List<BuiltInFn> =
     { name = fn "toBytes" 0
       typeParams = []
       parameters = [ Param.make "str" TString "" ]
-      returnType = TBytes
+      returnType = (TList(TUInt8))
```
Comment (pbiggar) [inline backend/src/BuiltinExecution/Libs/String.fs:397]:
```suggestion
      returnType = TList TUInt8
```

File: backend/src/BuiltinExecution/Libs/String.fs [hunk 5]
```diff
@@ -411,13 +410,14 @@ let fns : List<BuiltInFn> =
 
     { name = fn "fromBytesWithReplacement" 0
       typeParams = []
-      parameters = [ Param.make "bytes" TBytes "" ]
+      parameters = [ Param.make "bytes" (TList(TUInt8)) "" ]
```
Comment (pbiggar) [inline backend/src/BuiltinExecution/Libs/String.fs:413]:
```suggestion
      parameters = [ Param.make "bytes" (TList TUInt8) "" ]
```

**Review by pbiggar — APPROVED** [review]
This is great!

**Comment by pbiggar** [comment]
I made two suggestions, and they apply largely across the commit, which is that there are excess parentheses eg `(X(Y))` can always be `(X Y)` and can sometimes be just `X Y`. Feel free to do it as a followup, if not it's not a big deal.

**Pull request merged**